crates/replication/src/lib.rs

```diff
@@ -278,4 +278,26 @@ impl Writer {
 
         Ok(rows_affected)
     }
+
+    pub async fn query(
+        &self,
+        sql: &str,
+        params: impl Into<pb::query::Params> + Send,
+    ) -> anyhow::Result<pb::ResultRows> {
+        let (write_frame_no, rows) = self.client.query(sql, params.into()).await?;
+
+        tracing::trace!(
+            "statment executed on remote waiting for frame_no: {}",
+            write_frame_no
+        );
+
+        self.frame_no_notifier
```

Comment:
that's the bit we don't want, but I see that it was already fixed as part of the execute_batch PR right?